pkg/models/dashboards_public.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+var (
+	ErrPublicDashboardFailedGenerateUniqueUid = DashboardErr{
+		Reason:     "Failed to generate unique dashboard id",
+		StatusCode: 500,
+	}
+	ErrPublicDashboardNotFound = DashboardErr{
+		Reason:     "Public dashboard not found",
+		StatusCode: 404,
+		Status:     "not-found",
+	}
+	ErrPublicDashboardIdentifierNotSet = DashboardErr{
+		Reason:     "No Uid for public dashboard specified",
+		StatusCode: 400,
+	}
+)
+
+type PublicDashboardConfig struct {
+	IsPublic        bool            `json:"isPublic"`
+	PublicDashboard PublicDashboard `json:"publicDashboard"`
+}
+
+type PublicDashboard struct {
```

Comment:
```suggestion
type PublicDashboardConfig struct {
```

Rename this struct to PublicDashboardConfig, and remove the PublicDashboardConfig with IsPublic, which should simplify the code a bit. I cannot see any use for the object wrapping PublicDashboardConfig with IsPublic, makes the code harder to read & understand.  

When would there exist a PublicDashboardConfig for a dashboard that is not public? that seems like an error state. If we want to have that state IsPublic should be a property on PublicDashboardConfig

Reply:
The intent of having config and a flag is the scenario where someone configures a public dashboard and publishes it, but then later decides to disable the public dashboard (turning the isPublic flag off). 

In this state we can still save the public dashboard configuration they have set (time range, variable selection, refresh rate, etc...) but still disable the public dashboard. 

This saving is convenient for when they go to reenable the public dashboard, all their config would still be there.

Reply:
I don't think that would simplify things. As of now, if we need to do something with a specific dashboard, we have a row with a UID that we can reference it by. If we want to support slugs or other special keys for URLs, those can be added and supported incrementally. Committing to a json object and needing to enforce that schema in the frontend and backend will be more difficult to manage and harder to backfill should needs change.

Reply:
@jalevin not sure I understand your comment. How does this relate to the IsPublic field? if that is part of the dashboard_public_config table, does that not simplify things?  

> As of now, if we need to do something with a specific public dashboard, we have a row with a UID that we can reference it by. 

Yes, I am not against a uid that represents the dashboard public config. But you also have a public flag on the dashboard, that seems to complicate things, requiring updating two entities.

Reply:
Based on comments above, we'll move this to the public dashboard row.